docs/tutorials/vision/large_scale_classification.md

```diff
@@ -0,0 +1,247 @@
+    c=`basename $i`
+    echo $c
+    mkdir -p ${VAL_ROOT}/$c
+    for j in `ls $i/*.JPEG | shuf | head -n 50`; do
+        mv $j ${VAL_ROOT}/$c/
+    done
+done
+```
+
+### Pack images into record files
+While MXNet can read image files directly, it is recommended to pack the image files into a recordIO file for increased performance. MXNet provides a tool (tools/im2rec.py) to do this. To use this tool, MXNet and OpenCV’s python module needs to be installed in the system. OpenCV’s python module can be installed on Ubuntu using the command `sudo apt-get install python-opencv`.
+
+Set the environment variable `MXNET` to point to the MXNet installation directory and `NAME` to the name of the dataset. We assume MXNet is installed at `~/mxnet`
+
+```
+MXNET=~/mxnet
+NAME=full_imagenet_500_filtered
+```
+
+To create the recordIO files, we first create a list of images we want in the recordIO files and then use `im2rec` to pack images in the list into recordIO files. We create this list in `train_meta`. Training data is around 1TB. We split it into 8 parts, with each part roughly 100 GB.
+
+```
+mkdir -p train_meta
+python ${MXNET}/tools/im2rec.py --list True --chunks 8 --recursive True \
+train_meta/${NAME} ${ROOT}
+```
+
+We then resize the images such that the short edge is 480 px and pack the images into recordIO files. Since most of the work is disk I/O, we use multiple (16) threads to get the work done faster.
+
+```
+python ${MXNET}/tools/im2rec.py --resize 480 --quality 90 \
+--num-thread 16 train_meta/${NAME} ${ROOT}
+```
+
+Once done, we move the rec files into a folder named train.
+
+```
+mkdir -p train
+mv train_meta/*.rec train/
+```
+
+We do similar preprocessing for the validation set
+
+```
+mkdir -p val_meta
+python ${MXNET}/tools/im2rec.py --list True --recursive True \
+val_meta/${NAME} ${VAL_ROOT}
+python ${MXNET}/tools/im2rec.py --resize 480 --quality 90 \
+--num-thread 16 val_meta/${NAME} ${VAL_ROOT}
+mkdir -p val
+mv val_meta/*.rec val/
+```
+
+We now have all training and validation images in recordIO format in `train` and `val` directories respectively. We can now use these `.rec` files for training.
+
+## Training
+
+[ResNet](https://arxiv.org/abs/1512.03385) has shown its effectiveness on ImageNet competition. Our experiments also [reproduced](https://github.com/tornadomeet/ResNet) the results reported in the paper. As we increase the number of layers from 18 to 152, we see steady improvement in validation accuracy. Given this is a huge dataset, we will use Resnet with 152 layers.
+
+Due to the huge computation complexity, even the fastest GPU nowadays needs more than one day for a single pass of the data. We often need tens of epochs before the training converges to good validation accuracy. While we can use multiple GPUs in a machine, number of GPUs in a machine is often limited to 8 or 16. For faster training, in this tutorial, we will use multiple machines each using multiple GPUs to train the model.

```

Comment:
computation -> computational
Remove nowadays. It reads fine without it.
each using multiple GPUs -> each containing multiple GPUs